test/test_image.rb

```diff
@@ -0,0 +1,205 @@
+    end
+    assert_screenshot_matches win, 'retro.png'
+
+    # with tileable: true
+    tileable_tiles = Gosu::Image.load_tiles(media_path('rotation.png'), 50, 50, tileable: true)
+    win = ScreenshotWindow.draw(250, 100) do
+      tileable_tiles[0].draw(-80, 0, 0, scale_x: 4.0, scale_y: 4.0)
+      tiles[0].draw(50, 0, 0, scale_x: 4.0, scale_y: 4.0, mode: :add)
+
+      tileable_tiles[1].draw(0, 0, 0, scale_x: 0.5, scale_y: 0.5)
+      tiles[1].draw(50, 0, 0, scale_x: 0.5, scale_y: 0.5)
+    end
+    assert_screenshot_matches win, 'tileable.png', 0.98
+  end
+
+  def test_new_with_rect
+    Dir.chdir media_path do
+      Dir.mktmpdir do |tmpdir|
+        output_filename = File.join(tmpdir, 'new_with_rect.bmp')
+
+        image = Gosu::Image.new(media_path('inserted.png'), rect: [0, 0, 25, 25])
+        image.save(output_filename)
+        assert_equal_files 'triangle-25.bmp', output_filename
+      end
+    end
+  end
+
+  def test_subimage
+    win = ScreenshotWindow.draw(50, 50) do |w, h|
+      c = 0xff_ffffff
+      @image ||= Gosu::Image.new(media_path('triangle-50.png'))
+      @subimage ||= @image.subimage(15, 15, 20, 20)
+      @image.draw(0, 0, 0)
+      @subimage.draw(25, 25, 0, angle: 180)
+    end
+    assert_screenshot_matches win, 'trintrang.png'
+  end
+
+  #----------- Drawing an image
+
+  def test_draw_with_opts
+    win = ScreenshotWindow.draw(50, 50) do
+      @image ||= Gosu::Image.new(media_path('triangle-50.png'))
+      @image.draw(0, 0, 0, scale_x: 1, scale_y: 1, color: 0xff_ffffff, mode: :default)
+    end
+    assert_screenshot_matches win, 'triangle-50.png'
+  end
+
+  def test_draw_args_for_compat
+    win = ScreenshotWindow.draw(50, 50) do
+      @image ||= Gosu::Image.new(media_path('triangle-50.png'))
+      @image.draw(0, 0, 0, 1, 1, 0xff_ffffff, :default)
+    end
+    assert_output("", /DEPRECATION WARNING:/) do
+      assert_screenshot_matches win, 'triangle-50.png'
+    end
+  end
+
+  def test_draw_with_angle
+    win = ScreenshotWindow.draw(100, 100) do
+      @image ||= Gosu::Image.new(media_path('triangle-50.png'))
+      @image.draw(25, 25, 0, angle: 90)
+      @image.draw(50, 25, 0, angle: 180, center_x: 1.0)
+      @image.draw(50, 50, 0, angle: 215, center_y: 1.0)
+      @image.draw(50, 50, 0, center_x: -1.0, center_y: -1.0, scale_x: 0.5, scale_y: 0.5, color: 0xff_ffffff, mode: :default)
+    end
+    assert_screenshot_matches win, 'rotation.png'
+  end
+
+  def test_draw_as_quad
+    win = ScreenshotWindow.draw(100, 100) do |w, h|
+      c = 0xff_ffffff
+      @image ||= Gosu::Image.new(media_path('triangle-50.png'))
+      @image.draw_as_quad(0, 0, c, w, 0, c, w, h, c, w-35, 15, c, 0, :default)
+
+      # BUG: Drawing the quad from bottom to top is buggy (last Vertex is always wrong)
+      # should     4--3
+      # be:       /    \
+      #          1------2
+      @image.draw_as_quad(0, w, c, w, h, c, w-15, h/2, c, 15, h/2, c, 0, :default)
```

Comment:
Not sure about this one - But the one vertex is always off? I tried: 
1-2  | 3-4 | 1-2
4-3  | 1-2 | 3-4

Reply:
Update - might be an Linux thingy. Because thats the only green system on CI - every other platform failed this test (with my wrong image)....

Reply:
Shouldn't the first parameters be `0, h, c`?